---
title: C28198
ms.date: 11/04/2016
ms.prod: visual-studio-dev15
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C28198
helpviewer_keywords:
- C28198
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 34adf5ffa6701e906285274f6928d987f1f526ae
ms.sourcegitcommit: e13e61ddea6032a8282abe16131d9e136a927984
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 04/26/2018
ms.locfileid: "31889697"
---
# <a name="c28198"></a>C28198
avertissement C28198 : éventuellement une fuite de mémoire en raison d’une exception.

 Cet avertissement indique que la mémoire allouée n'est pas libérée après qu’une exception est levée. L’instruction à la fin du chemin d’accès peut lever une exception. La mémoire a été passée à une fonction qui peut avoir enregistré une copie à libérer plus tard.

 Cet avertissement est très similaire à l’avertissement [C28197](../code-quality/c28197.md). Les annotations sont recommandées pour une utilisation avec l’avertissement [C28197](../code-quality/c28197.md) peuvent également être utilisés ici.

## <a name="example"></a>Exemple
 L'exemple de code suivant génère cet avertissement :

```
char *p1 = new char[10];
char *p2 = new char[10];

test(p1);   // does not save a copy of p

delete[] p2;
delete[] p1;
```

 L'exemple de code suivant évite cet avertissement :

```
char *p1 = new char[10];
char *p2 = NULL;

test(p1);   // does not save a copy of p
try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
```